project/ntut-xuan/README.rst

```diff
@@ -0,0 +1,198 @@
+image bytes buffer to these classes and got the byte buffer of processed
+image as result.
+
+As we got the byte buffer of processed image, we can transfer it into
+images and process shredding and voting.
+
+API Description
+---------------
+
+Since having ``GaussianBlur`` and ``OtsuMethod`` class in C++, it should
+have constructor and destructor, which can passed the image byte buffer
+into constructor and release the dynamic resource (if any) in
+destructor. These class should have some customize parameter to
+determine the particles, e.g. may passed :math:`\sigma` in
+``GaussianBlur`` to determine Gaussian filter kernel and threshold
+:math:`t` in ``OtsuMethod``.
+
+Moreover, we may provide some function on ``GaussianBlur`` and
+``OtsuMethod`` to validate these classes is work correctly. For example,
+may having ``GetGaussianFilterKernel`` in ``GaussianBlur`` and it should
+return the matrix of Gaussian filter kernel. With these function, we can
+validate our ``GaussianBlur`` work correctly.
+
+Pending Task
+------------
+
+-  [2] Properly Command-line Interface to input image and got size of
+   shred pieces.
+
+   e.g. ``./polygondust --input image.png --particles 5`` means input
+   ``image.png`` and size of rectangle should be 5x5.
+
+-  ``Pybind11`` for convert the C/C++ class into python methods.
+
+   -  [2] Working on ``GaussianBlur`` class and make sure we can reach
+      all public function in this class in Python script.
+   -  [2] Working on ``OtsuMethod`` class and make sure we can reach all
+      public function in this class in Python script.
+
+-  Implement the feature on ``GaussianBlur`` and ``OtsuMethod`` class
+
+   -  [7] Working on ``GaussianBlur`` class and trying to validate the
+      ``GaussianBlur`` working properly.
+   -  [7] Working on ``OtsuMethod`` class and trying to validate the
+      ``OtsuMethod`` working properly.
+
+-  [3] Implement the shred feature on Python script.
+
+-  [3] Implement voting feature on Python script.
+
+``[X]`` Means the weight of the task, it used to measure the task
+progress.
+
+Engineering Infrastructure
+--------------------------
+
+In this project, we have these infrastructure to make sure our project
+can develop properly.
+
+-  Version Control: Git to maintaince the version control and GitHub to
+   handle the repository.
+-  Build Tools (Package Tools): CMake for C/C++.
+-  Testing framework: GoogleTest for C/C++ and Pytest for Python.
+-  The classes and function have its unit test.
+-  Coverage Supervise Tools: Codecov to supervise our testing coverage
+   in these test cases.
+-  Documentation: Nice ``README.md``
+
+Schedule
+--------
+
+.. list-table:: Title
+   :widths: 25 25 50
+   :header-rows: 1
+
+ * - Week
+   - Task 
+   - Progress (%)
+ * - 1
+   - System Design / Writing proposal
+   - 0% (00/26)
+ * - 2
+   - Create GitHub Project, trying to figure how to process image on C++, and how to process command line interface on C++
+   - 0% (00/26)
+ * - 3
```

Comment:
When do you plan to set up Github Actions and the initial set of automatic tests?

Reply:
I would like to keep all the commit should passed the test. Beside, I would like to add Codecov to supervise my test coverage, so it will need GitHub Actions.

Reply:
Codecov is not very useful, but you can try.  Please make testing early.